### ISSUE TITLE
Unable to edit fields - intermittent

### ISSUE THREAD

**Issue by sussexrick** [umbraco/Umbraco.Forms.Issues#2]
First logged as http://issues.umbraco.org/issue/CON-1561

We are repeatedly seeing an error with Umbraco Forms where fields become non-editable. When you click the cog icon for the field in the form designer, the overlay to edit the field doesn't appear. It might affect every field on a form, or just some. Oddly, the problem will correct itself but with no apparent pattern as to when, so you can't know when you'll be able to edit a form. Copying the form doesn't help as the copy has the same problem. Reordering fields doesn't help. When the problem is happening, other forms might be fine. The only workaround we have is to refresh repeatedly until it works. 

A visual clue that there's a problem when you're in the form designer is that the preview of the field doesn't load correctly, eg for a "Single choice" field you would normally see the prevalues listed as radio buttons, but they don't appear.

There are messages in the Chrome console which point to $scope.form being null as the immediate cause.

```
DependencyHandler.axd?s=L3Vt.....&t=Javascript&cdv=93518197:8 TypeError: Cannot read property 'pages' of undefined
    at Object.initForm (https://www.example.org/DependencyHandler.axd?s=L0Fwc...&t=Javascript&cdv=93518197:273:1029)
    at Object.scope.initForm (https://www.example.org/DependencyHandler.axd?s=L0Fwc...&t=Javascript&cdv=93518197:264:2705)
    at link (https:/www.example.org/DependencyHandler.axd?s=L0Fwc...

 <div class="umb-forms-designer clearfix ng-scope" ng-if="vm.ready &amp;&amp; vm.security" form="model" currentpage="vm.currentPage" fieldtypes="vm.fieldtypes" prevaluesources="vm.prevaluesources" security="vm.security">
```

```
TypeError: Cannot read property 'pages' of undefined
    at Object.scope.formHasFields (DependencyHandler.axd?s=L0Fwc...&t=Javascript&cdv=93518197:264)
    at DependencyHandler.axd?s=L3Vt...&t=Javascript&cdv=93518197:9
    at && (DependencyHandler.axd?s=L3Vt...&t=Javascript&cdv=93518197:14)
    at Object.t.constant [as get] (DependencyHandler.axd?s=L3Vt...&t=Javascript&cdv=93518197:9)
    at Object.$digest (DependencyHandler.axd?s=L3Vt...&t=Javascript&cdv=93518197:10)
    at Object.$apply (DependencyHandler.axd?s=L3Vt...&t=Javascript&cdv=93518197:10)
    at j (DependencyHandler.axd?s=L3Vt...&t=Javascript&cdv=93518197:10)
    at r (DependencyHandler.axd?s=L3Vt...&t=Javascript&cdv=93518197:11)
    at XMLHttpRequest.w.onreadystatechange (DependencyHandler.axd?s=L3Vt...&t=Javascript&cdv=93518197:12)
```

```
DependencyHandler.axd?s=L3Vt...&t=Javascript&cdv=93518197:8 TypeError: Cannot read property 'settings' of undefined
    at DependencyHandler.axd?s=L0Fwc...&t=Javascript&cdv=93518197:266
    at Object.n [as forEach] (DependencyHandler.axd?s=L3Vt...&t=Javascript&cdv=93518197:2)
    at Object.scope.setFieldType (DependencyHandler.axd?s=L0Fwc...&t=Javascript&cdv=93518197:266)
    at a.scope.openFieldSettings (DependencyHandler.axd?s=L0Fwc...&t=Javascript&cdv=93518197:265)
    at DependencyHandler.axd?s=L3Vt...&t=Javascript&cdv=93518197:9
    at DependencyHandler.axd?s=L3Vt...&t=Javascript&cdv=93518197:41
    at a.$eval (DependencyHandler.axd?s=L3Vt...&t=Javascript&cdv=93518197:10)
    at a.$apply (DependencyHandler.axd?s=L3Vt...&t=Javascript&cdv=93518197:10)
    at HTMLAnchorElement.<anonymous> (DependencyHandler.axd?s=L3Vt...&t=Javascript&cdv=93518197:41)
    at HTMLAnchorElement.dispatch (DependencyHandler.axd?s=L3Vt...&t=Javascript&cdv=93518197:1)
```

**Comment by pijemcolu** [comment]
Thanks for taking the time to re-create the issue @sussexrick. What Umbraco Forms & CMS versions do you see this in?

**Comment by sussexrick** [comment]
We see this with Umbraco 7.6.13 and Forms 6.0.8.

**Comment by sussexrick** [comment]
We have just upgraded to Umbraco 7.14.0 and Forms 7.1.1 and are still seeing this problem.

**Comment by Ultradata-CXP** [comment]
Thanks @sussexrick.  We also experiencing this problem and are currently investigating.  the problem appears to be limited to a single form within a suite of forms that we have on the install.  Unlike team members I've been unable to view fields/access config with multiple refreshes so far.

**Comment by Ultradata-CXP** [comment]
We find that if we clone the form and remove the workflow, the form becomes editable.

**Comment by Alkaersig** [comment]
We are seeing this problem as well on Umbraco 7.14.0 with Forms 7.1.5. When using the tree to navigate to a form containing workflows, the error is triggering. Refreshing with F5 loads the same form without errors.

**Comment by bipinjz** [comment]
We found that the issue happens when a form has large number of complex workflows and form gets initialized before data is received. The quick workaround we used is to reinitialize the form  after getting the response.data in getWithWorkflowsByGuid  in umbraco.forms.js.  This will still show the original errors but form will work.

Hope that helps!

**Comment by Alkaersig** [comment]
Thanks @bipinjz. That makes sense, however it still sounds like a bug that should be fixed in the forms implementation.

**Comment by sussexrick** [comment]
Our forms have only a couple of simple workflows, so complex workflows aren't the cause.

**Comment by Alkaersig** [comment]
We only have the default workflow and a workflow that fetches a list of audiences from MailChimp (happens very fast), but the MailChimp is enough to trigger the error. When doing a full refresh it loads without errors.

**Comment by Deba22** [comment]
I'm still facing this issue.Has this been fixed?
I'm using Umbraco 7.12.3 and Forms 7.0.4.
Any workaround to fix this?

**Comment by Alkaersig** [comment]
Yes, unfortunately. It is still an issue even if it's an old one.

**Comment by Deba22** [comment]
@Alkaersig Thanks for the confirmation.
Is there any workaround for this?

**Comment by Alkaersig** [comment]
@Deba22 there was posted a description of a solution @bipinjz a few posts back. Don't know if that is any help or if he could describe it a little more detailed.

**Comment by nul800sebastiaan** [comment]
Fixed in #232
